rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,633 @@
+        - name: parent
+          in: query
+          description: Optional parent namespace under which to list namespaces. When empty, list top-level namespaces.
+          required: false
+          schema:
+            type: string
+          example: ?parent=accounting.tax
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListNamespacesResponse'
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found (Parent namespace does not exist)
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description: Create a namespace, with an optional set of properties. The server might also add properties, such as last_modified_time etc.
+      operationId: createNamespace
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateNamespaceRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        '409':
+          $ref: '#/components/responses/NamespaceAlreadyExistsResponse'
+
+  /namespaces/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: Load the metadata properties for a namespace
+      operationId: loadNamespaceMetadata
+      description: Return all stored metadata properties for a given namespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetNamespaceResponse'
+        '404':
+          $ref: '#/components/responses/NoSuchNamespaceResponse'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a namespace exists
+      operationId: namespaceExists
+      description: Check if a namespace exists.
+      responses:
+        '200':
+          description: OK - Namesapce exists
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a namespace from the catalog. Namespace must be empty.
+      operationId: dropNamespace
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergResponseObject'
+              example: { "data": { "dropped": true } }
+        '401':
+          description: Unauthorized
+        '404':
+          description: Not Found
+
+
+  /namespaces/{namespace}/properties:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    post:
```

Comment:
The consensus was to try to keep things that aren't too much cognitive overhead for people.

Does `application/merge-patch+json` content-type encompass that? I would personally argue yes.That if somebody were sitting at their terminal or with an HTTP request GUI o whatever and sending requests, including the `application/merge-patch+json` content-type would likely be a source of frustration and something that needs to be looked up regularly.

Others might disagree though.

Reply:
I do like the general form of the merge-patch, but looking at libraries listed here, support does not seem to be universal: http://jsonpatch.com/

Reply:
As for the complexity, I mean having all three of `POST` / `PUT` / `PATCH`, especially as `PATCH`.

There have been so far people in support of one or the other, but the consensus was to avoid putting too many things into one route, so as to make things a bit more self documenting. Particularly when `PATCH` could refer to any number of potential patches but so far it only corresponds to one.

Just for insight / clarity as to what has been discussed somewhat so far (both here ... somewhere in the PR discussion .... and elsewhere offline etc).